src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/DirectoryBasedStoreProvider.cs

```diff
@@ -298,24 +302,7 @@ public void Add(ICertificatePal certPal)
                     stream.Write(pkcs12, 0, pkcs12.Length);
                 }
 
-#if DEBUG
-                // Verify that we're creating files with u+rw and o-rw, g-rw.
-                const Interop.Sys.Permissions requiredPermissions =
-                    Interop.Sys.Permissions.S_IRUSR |
-                    Interop.Sys.Permissions.S_IWUSR;
-
-                const Interop.Sys.Permissions forbiddenPermissions =
-                    Interop.Sys.Permissions.S_IROTH |
-                    Interop.Sys.Permissions.S_IWOTH |
-                    Interop.Sys.Permissions.S_IRGRP |
-                    Interop.Sys.Permissions.S_IWGRP;
-
-                Interop.Sys.FileStatus stat;
-
-                Debug.Assert(Interop.Sys.Stat(destinationFilename, out stat) == 0);
-                Debug.Assert((stat.Mode & (int)requiredPermissions) == (int)requiredPermissions, "Created PFX has insufficient permissions to function");
-                Debug.Assert((stat.Mode & (int)forbiddenPermissions) == 0, "Created PFX has too broad of permissions");
-#endif
+                EnsureFilePermissions(destinationFilename, userId);
```

Comment:
Sorry for all the comment, but one more thought... if you do the latter, rather than doing stat on the destinationFilename, I think you should be able to do fstat on the stream's SafeFileHandle.DangerousGetHandle(), which is the file descriptor.  That should ensure that you're actually getting the permissions of the file you're writing to, in case someone somehow managed to unlink and then create a new file between the time that you opened the file and did the permissions check.